app/uk/gov/hmrc/eusubsidycompliancefrontend/models/emailSend/EmailParameters.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eusubsidycompliancefrontend.models.emailSend
+
+import ai.x.play.json.Jsonx
+import play.api.libs.json.{Format, JsObject, JsResult, JsValue, Json, OFormat}
+import uk.gov.hmrc.eusubsidycompliancefrontend.models.types.{EORI, UndertakingName, UndertakingRef}
+
+sealed trait EmailParameterType
+
+object EmailParameterType {
+
+  case object  SingleEORI extends EmailParameterType
+  case object DoubleEORI extends EmailParameterType
+  case object DoubleEORIAndDate extends EmailParameterType
+  case object  SingleEORIAndDate extends EmailParameterType
+
+  import ai.x.play.json.SingletonEncoder.simpleName
```

Comment:
Out of interest, what do these extensions give us over standard play JSON?

Reply:
hope this helps [link](https://index.scala-lang.org/bizzabo/play-json-extensions/play-json-extensions/0.42.1?target=_2.12)

Reply:
Thanks, will check it out. :)